File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        
        Arrays.sort(phone_book);
        
        for (int i = 0; i < phone_book.length - 1; i++) {
            String prePhoneNumber = phone_book[i];
            String nextPhoneNumber = phone_book[i+1];
            if (nextPhoneNumber.startsWith(prePhoneNumber)) {
                answer = false;
                break;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    phone_book.sort()
    
    for i in range(0, len(phone_book) - 1):
        n1 = phone_book[i]
        n2 = phone_book[i + 1]
        
        if n1 in n2 and n1[0] == n2[0]:
            answer = False
            break
    return answer